atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/transaction/impl/TransactionLocksManagerTest.java

```diff
@@ -24,55 +24,73 @@
-public final class ImmutableTimestampLockManagerTest {
+@ExtendWith(MockitoExtension.class)
+public final class TransactionLocksManagerTest {
     private static final LockToken DEFAULT_IMMUTABLE_TIMESTAMP_LOCK_TOKEN = LockToken.of(UUID.randomUUID());
     private static final LockToken DEFAULT_COMMIT_LOCK_TOKEN = LockToken.of(UUID.randomUUID());
 
+    @Mock
+    private LockUnlocker unlocker;
+
+    @Mock
+    private LockValidityChecker validityChecker;
+
     @ParameterizedTest(name = "{2}")
     @MethodSource("lockManagerConfigurations")
     public void returnsNothingExpiredWhenAllLocksStillValid(
             Optional<LockToken> immutableTimestampLock, Optional<LockToken> userCommitLock, String testDescription) {
-        LockValidityChecker validityChecker = mock(LockValidityChecker.class);
-        ImmutableTimestampLockManager immutableTimestampLockManager =
-                new ImmutableTimestampLockManager(immutableTimestampLock, validityChecker);
+        TransactionLocksManager transactionLocksManager =
+                new TransactionLocksManager(immutableTimestampLock, validityChecker, unlocker);
 
-        when(validityChecker.getStillValidLockTokens(anySet()))
-                .thenAnswer(invocation -> invocation.getArguments()[0]);
+        if (immutableTimestampLock.isPresent() || userCommitLock.isPresent()) {
+            when(validityChecker.getStillValidLockTokens(anySet()))
+                    .thenAnswer(invocation -> invocation.getArguments()[0]);
+        }
+        userCommitLock.ifPresent(transactionLocksManager::registerLock);
 
-        assertThat(immutableTimestampLockManager.getExpiredImmutableTimestampAndCommitLocks(userCommitLock))
+        assertThat(transactionLocksManager.getExpiredImmutableTimestampAndCommitLocks())
                 .isEmpty();
     }
 
     @ParameterizedTest(name = "{1}")
     @MethodSource("lockManagerConfigurationsWithPresentUserCommitLock")
     public void returnsLocksThatWereCheckedWhenGettingFullSummaryAndStillValid(
             Optional<LockToken> immutableTimestampLock, String testDescription) {
-        LockValidityChecker validityChecker = mock(LockValidityChecker.class);
-        ImmutableTimestampLockManager immutableTimestampLockManager =
-                new ImmutableTimestampLockManager(immutableTimestampLock, validityChecker);
+        TransactionLocksManager transactionLocksManager =
+                new TransactionLocksManager(immutableTimestampLock, validityChecker, unlocker);
 
         when(validityChecker.getStillValidLockTokens(anySet()))
                 .thenAnswer(invocation -> invocation.getArguments()[0]);
 
-        assertThat(immutableTimestampLockManager.getExpiredImmutableTimestampAndCommitLocksWithFullSummary(
-                        DEFAULT_COMMIT_LOCK_TOKEN))
+        Set.of(DEFAULT_COMMIT_LOCK_TOKEN).forEach(transactionLocksManager::registerLock);
```

Comment:
nit: curious about `Set.of(...)` here and later in the test, where the set has 1 element - is the goal to extend this to account for other locks that can be registered?

Reply:
Oh I was just rushing, sorry. Lemme cleanup.

Reply:
Fixed